### PR TITLE
Create recipe react

### PR DESCRIPTION
**Pull request by EmilianoC02**
Created front end for recipe landing, recipe lists, grocery lists, and stock lists. Connected all components to back end. I had to tinker with the main.py API routes to make them work and update the delete_item functions from Stock.py and GroceryList.py to where if the current list is empty, it'll delete that list from the database. This still works because if a user attempts to add an item and a list for that user doesn't exist yet, it'll create a new one. Also, I did this because it allowed me to fix a rendering problem I had with the list items. If the list existed but contained nothing, a random delete item button would appear that would bug the program if clicked. 

Still need to talk to Grayson about how the grocery list is going to populate the stock list when users "buy" the items. Also, forcing the user to input the recipe data the way it's supposed to go into the database is super complicated. For now, I just put some placeholder items that show exactly how the user should input their data. Will need to talk to Grayson about how we can ensure all data is inputted into the database the way it should be. 

SAAS4N4L

### CODE REVIEW

**Comment by ryanmercado** [comment]
Lets get @WilliamSterner to review the changes to his flask app, I added him as a reviewer

**Comment by ryanmercado** [comment]
Also it seems like we need to make some group decisions about how data is being inputted from the frontend

File: backend/handleCreateAccount.py
[the remark this answers is not in the document]
Comment (Grayson34) [reply backend/handleCreateAccount.py:1]: well i guess u removed pointless stuff anyway. Looks fine

File: backend/recipe/GroceryList.py [hunk 1]
```diff
@@ -60,8 +60,11 @@ def delete_item(id, item):
                     continue
                 else:
                     shaved_grocery_list.append(existing_item)
-            final_list = items_string = ', '.join(shaved_grocery_list)
-            cursor.execute('UPDATE grocery_list SET grocery_list = ? WHERE usr_id = ?', (final_list, id))
+            final_list = ', '.join(shaved_grocery_list)
+            if final_list:
```
Comment (Grayson34) [inline backend/recipe/GroceryList.py:64]: why cant we just leave empty list, does it break?

File: backend/recipe/__pycache__/GroceryList.cpython-310.pyc
Comment (Grayson34) [inline backend/recipe/__pycache__/GroceryList.cpython-310.pyc:1]: dont push these. remove them before comitting. hope it doesnt break the pr

File: backend/recipe/Stock.py [hunk 2]
```diff
@@ -62,8 +62,10 @@ def delete_item(id, del_item):
                 else: 
                     new_stock.append(item)
             items_string = ', '.join(new_stock)
-            cursor.execute('UPDATE curr_stock SET stock_list = ? WHERE usr_id = ?', (items_string, id))
-            
+            if items_string:
+                cursor.execute('UPDATE curr_stock SET stock_list = ? WHERE usr_id = ?', (items_string, id))
+            else:
```
Comment (Grayson34) [inline backend/recipe/Stock.py:67]: same question here as above

File: student-hub-app/package.json
Comment (Grayson34) [inline student-hub-app/package.json:1]: whats this file

File: backend/recipe/GroceryList.py (hunk 1 above)
Comment (EmilianoC02) [reply backend/recipe/GroceryList.py:64]: Yes. If left empty, the frontend renders an empty bullet point with a delete button that can mess things up. The add recipe or grocery or stock item functions create a new list if there's no list associated with that usr_id, so this works.

**Review by ryanmercado — APPROVED** [review]
Lookin A-okay

**Pull request merged**